packages/migration/src/migrations/v0047_fixBrokenTemplates.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import type { ElementId } from '@googleforcreators/elements';
+
+/**
+ * Internal dependencies
+ */
+import type { AnimationV0 } from '../types';
+import type {
+  PageV46,
+  StoryV46,
+  UnionElementV46,
+} from './v0046_removeRedundantScalingProperties';
+
+export type StoryV47 = StoryV46;
+export type PageV47 = PageV46;
+export type UnionElementV47 = UnionElementV46;
+
+function fixBrokenTemplate({ pages, ...rest }: StoryV46): StoryV47 {
+  return {
+    pages: pages.map(fixBrokenPage),
+    ...rest,
+  };
+}
+
+function fixBrokenPage({ elements, animations, ...rest }: PageV46): PageV47 {
+  const elementIds = elements.map(({ id }) => id);
+
+  return {
+    elements: fixElementOrder(elements),
+    ...(animations
+      ? {
+          animations: (animations as AnimationV0[]).filter((animation) =>
```

Comment:
`[@typescript-eslint/no-unnecessary-type-assertion] This assertion is unnecessary since it does not change the type of the expression.`